### DISCUSSION TITLE
"publish_from" - "publish_to" as optional system fields

### DISCUSSION THREAD

**Discussion by Michael-Nussbaumer** [directus/directus#4347] [Feature Requests]
I would like to see `publish_from`  - `publish_to` as optional system fields in addition to the `status`. 

This should be a common use case for items like news, to be published on an exact time in the future. Or taking something offline after for example an offer has expired.

You can kind of build this by yourself by creating the fields and filter it on your requests. But this can be some major problem in some cases. Or you have to build custom endpoints for that. But it would be neat to handle this internally.

Lets take this real life example: Tesla announced to invest 1.5 billion in bitcoin. If someone with some knowledge removing the filter from the api call, he could have gotten this info before publication in this case. So you wouldn't want someone to getting this knowledge before publication, as it also might follow up with some legal problems.

**Comment by rijkvanzanten** [comment]
> But this can be some major problem in some cases.

What kind of problem?

> If someone with some knowledge removing the filter from the api call, he could have gotten this info before publication in this case. So you wouldn't want someone to getting this knowledge before publication, as it also might follow up with some legal problems.

This is something you can easily handle through permissions 🙂  Set the read permissions for the (public) role to only allow read for articles where publish_from <= $NOW.

**Reply by Michael-Nussbaumer** [reply]
Didn't had that in my mind. :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
